README.md

```diff
@@ -36,38 +36,109 @@ The enrolling repository uses two scripts to stay synced with this repository:
    automatic PRs.
 
+Enrollment also creates a file `.make_codeclimate`, which it uses to generate
+your project's custom `.codeclimate.yml`. You'll edit this rather than
+editing `.codeclimate.yml` directly. This lets `conjur-project-config`
+automatically manage and keep current the base portion of `.codeclimate.yml`
+shared among all projects. See "How To Enroll" below for more.
+
 ## How to Enroll
 
-First embed this repository into your "enrolling" repository as a git submodule:
+### Step 1: Add git submodule
+
+After creating a new branch, embed this repository into your project's
+repository as a git submodule:
 
 ```bash
 git submodule add -b main 'https://github.com/cyberark/conjur-project-config.git'
 ```
 
-Next update the settings by running this command from the root of the enrolling
-repository:
+### Step 2: Run `update-settings`
+
+Update the settings by running the following command from the root of
+the enrolling repository:
 
 ```bash
 # Note: Run from root of enrolling repository
 ./conjur-project-config/update-settings
 ```
 
-Add a commit with a subject like `Import settings for conjur-project-config`.
+You can run `git status` to see what changes the script made.
+
+### Step 3: Configure `.make_codeclimate` and re-run `update_settings`
+
+`.make_codeclimate` (created by the previous step) lets you customize
+`.codeclimate.yml` for your repository.
+
+To enable support in `.make_codeclimate` for ruby or rails projects,
+uncomment the relevant lines. `.make_codeclimate` is just bash, and can
+output arbitrary yaml if your project needs to enable other CodeClimate
+plugins. You can also open a pull request on this project to add new
+templates for other languages.
 
-Finally, add or update the `.github/workflow` files:
+**Once enrolled in `conjur-project-config`, you should never edit
+`.codeclimate.yml` directly anymore. Instead, edit `.make_codeclimate`.**
+
+**Any time you edit `.make_codeclimate`, you'll need to re-run `update-settings`
+to regenerate `.codeclimate.yml` based on your changes.**
+
+### Step 4: Copy `.github/workflow` files
+
+To add the `.github/workflow` files that will automatically create pull
+requests in your project when `conjur-project-config` settings change in the
+future:
 
 ```bash
 # Note: Run from root of enrolling repository
 ./conjur-project-config/update-workflow
 ```
 
-Add a commit with a subject like `Enroll in automatic settings updates`.
+### Step 5: Commit your changes and open a pull request
+
+Run:
+
+```bash
+SKIP_GITLEAKS=YES git add . && git commit -m "Enroll in shared settings"
+```
+
+*Note: The `SKIP_GITLEAKS=YES` is needed because gitleaks doesn't play well
+with submodules.*
+
+Finally, push your branch to Github and open a new pull request. Once it's
+merged into `main`, your project will be enrolled in automatic settings
+updates: Whenever `conjur-project-config` settings change, a pull request
+will automatically open.
```

Comment:
Is it worth discussing that CodeClimate might flag a bunch of issues on this initial commit & it might need a GitHub admin to do the merge to ignore the issues initially?

Something like:
```suggestion
will automatically open.

CodeClimate may identify a number of new issues, and depending on how it is configured in this repository's settings, may block the PR from being merged because of these new issues. It may require a GitHub admin (an EM or Infra team member) to merge the PR and override the check. Once the PR is merged, CodeClimate will still track the issues but only block PRs on new issues in changes. 
```

Reply:
Change made, thanks.